Content.Shared/_DV/Projectiles/ProjectileForeignBodySystem.cs

```
@@ -0,0 +1,172 @@
+    }
+
+    private void OnEmbedProjectileHit(Entity<ProjectileForeignBodyComponent> ent, ref ProjectileHitEvent args)
+    {
+        if (!TryComp(ent, out ProjectileComponent? projectile))
+            return;
+
+        var evt = new CoefficientQueryEvent(SlotFlags.All);
+        RaiseLocalEvent(args.Target, evt);
+
+        var chance = ent.Comp.BaseChance * evt.DamageModifiers.Coefficients.GetValueOrDefault(ent.Comp.DamageType, 1);
+        if (_random.NextFloat() > chance)
+            return;
+
+        var ev = new ProjectileForeignBodyAttemptEvent(projectile.Shooter!.Value, projectile.Weapon!.Value, args.Target, ent);
+        RaiseLocalEvent(args.Target, ref ev);
+    }
+
+    private void OnSimpleInit(Entity<SimpleForeignBodyContainerComponent> ent, ref ComponentInit args)
+    {
+        _container.EnsureContainer<Container>(ent, ent.Comp.ContainerName);
+    }
+
+    private void OnSimpleEmbed(Entity<SimpleForeignBodyContainerComponent> ent, ref ProjectileForeignBodyAttemptEvent args)
+    {
+        if (_net.IsClient) // i yearn for predicted spawning
+            return;
+
+        if (!TrySpawnInContainer(args.Embedded.Comp.ForeignBody, ent, ent.Comp.ContainerName, out var spawned))
+            return;
+
+        var active = EnsureComp<ForeignBodyActivelyEmbeddedComponent>(spawned.Value);
+        active.ActiveAfter = _timing.CurTime + args.Embedded.Comp.EffectsBeginAfter;
+    }
+
+    private void OnSimpleEffect(Entity<SimpleForeignBodyContainerComponent> ent, ref ForeignBodyEffectsEvent args)
+    {
+        ApplyBodyPartEffects(ent, ent, args.Embedded);
+    }
+
+    private void OnBodyEmbed(Entity<BodyForeignBodyContainerComponent> ent, ref ProjectileForeignBodyAttemptEvent args)
+    {
+        if (_net.IsClient) // i yearn for predicted spawning
+            return;
+
+        var targetPart = _body.GetRandomBodyPart(args.Shooter) ?? TargetBodyPart.Torso;
```

use the shooters targeted part and only fall back to a random one if that fails